File: api/Enkl.Api/Controllers/OrganisationsController.cs
using Enkl.Api.Auth;
using Enkl.Api.Dtos;
using Enkl.Api.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Enkl.Api.Controllers;

[ApiController]
[Authorize(Policy = "OrgAdmin")]
[Route("api/organisations/me")]
public class OrganisationsController : ControllerBase
{
    private readonly OrganisationService _organisations;

    public OrganisationsController(OrganisationService organisations)
    {
        _organisations = organisations;
    }

    [HttpGet]
    public async Task<IActionResult> GetMyOrganisation()
    {
        var result = await _organisations.GetOrganisationAsync(User.OrgId());
        return result is null ? NotFound() : Ok(result);
    }

    [HttpPut("users/{userId:guid}/admin")]
    public async Task<IActionResult> SetUserAdmin(Guid userId, SetOrgAdminRequest request)
    {
        var ok = await _organisations.SetUserAdminAsync(User.OrgId(), userId, request.IsOrgAdmin);
        return ok ? NoContent() : NotFound();
    }

    [HttpPost("users")]
    public async Task<IActionResult> CreateUser(CreateUserRequest request)
    {
        var result = await _organisations.CreateUserAsync(User.OrgId(), request);
        return Ok(result);
    }

    [HttpPut("users/{userId:guid}/email")]
    public async Task<IActionResult> SetUserEmail(Guid userId, SetUserEmailRequest request)
    {
        var ok = await _organisations.SetUserEmailAsync(User.OrgId(), userId, request.EmailAddress);
        return ok ? NoContent() : NotFound();
    }

    [HttpGet("org-teams")]
    public async Task<IActionResult> GetOrgTeams()
    {
        return Ok(await _organisations.GetOrgTeamsAsync(User.OrgId()));
    }

    [HttpPut("default-password")]
    public async Task<IActionResult> SetDefaultNewUserPassword(SetDefaultNewUserPasswordRequest request)
    {
        var ok = await _organisations.SetDefaultNewUserPasswordAsync(User.OrgId(), request.Password);
        return ok ? NoContent() : NotFound();
    }
}

File: mariadb-api/src/Controllers/OrganisationsController.php
<?php

declare(strict_types=1);

namespace Enkl\Api\Controllers;

use Enkl\Api\Db\Database;
use Enkl\Api\Services\OrganisationService;
use Psr\Http\Message\ResponseInterface as Response;
use Psr\Http\Message\ServerRequestInterface as Request;

/** Ported from Controllers/OrganisationsController.cs — every action here requires OrgAdminMiddleware (see routes.php). */
final class OrganisationsController extends BaseController
{
    private function service(): OrganisationService
    {
        return new OrganisationService(Database::connection());
    }

    public function getMyOrganisation(Request $request, Response $response): Response
    {
        $org = $this->service()->getOrganisation($this->callerOrgId($request));
        return $org === null ? $this->notFound($response) : $this->json($response, $org);
    }

    public function setUserAdmin(Request $request, Response $response, array $args): Response
    {
        $body = $this->body($request);
        $ok = $this->service()->setUserAdmin($this->callerOrgId($request), $args['userId'], (bool) ($body['isOrgAdmin'] ?? false));
        return $ok ? $this->noContent($response) : $this->notFound($response);
    }

    public function createUser(Request $request, Response $response): Response
    {
        $result = $this->service()->createUser($this->callerOrgId($request), $this->body($request));
        return $this->json($response, $result);
    }

    public function setUserEmail(Request $request, Response $response, array $args): Response
    {
        $body = $this->body($request);
        $ok = $this->service()->setUserEmail($this->callerOrgId($request), $args['userId'], $body['emailAddress'] ?? null);
        return $ok ? $this->noContent($response) : $this->notFound($response);
    }

    public function getOrgTeams(Request $request, Response $response): Response
    {
        return $this->json($response, $this->service()->getOrgTeams($this->callerOrgId($request)));
    }

    public function setDefaultNewUserPassword(Request $request, Response $response): Response
    {
        $body = $this->body($request);
        $ok = $this->service()->setDefaultNewUserPassword($this->callerOrgId($request), (string) ($body['password'] ?? ''));
        return $ok ? $this->noContent($response) : $this->notFound($response);
    }
}
